server/src/main/java/io/unitycatalog/server/persist/utils/RepositoryUtils.java

```diff
@@ -0,0 +1,26 @@
+package io.unitycatalog.server.persist.utils;
+
+import io.unitycatalog.server.persist.PropertyRepository;
+import io.unitycatalog.server.persist.dao.PropertyDAO;
+import io.unitycatalog.server.utils.Constants;
+import org.hibernate.Session;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+public class RepositoryUtils {
+    public static <T> T attachProperties(T entityInfo, String uuid, String entityType, Session session) {
+        try {
+            List<PropertyDAO> propertyDAOList = PropertyRepository.findProperties(
+                    session, UUID.fromString(uuid), entityType);
+            Method setPropertiesMethod = entityInfo.getClass().getMethod("setProperties", Map.class);
```

Comment:
Can we make all entityInfo inherit from a common Interface such as `HasProperties` to avoid using reflection? We can do this as a followup.

Reply:
Noted.